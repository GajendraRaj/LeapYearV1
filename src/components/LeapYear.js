import React, { useState } from 'react';

function LeapYear() {
    const [state, setState] = useState({ year: 0 });

    const handleInputChange = (e) => {
        const year = e.target.value;
        
        setState(prevState => {
            return { ...prevState, year: year }
          });
      }

    const checkLeapYear = (year) => {
        if(year && year.toString().length > 3 && year.toString().length < 5) {
            const isLeapYear = ((year % 4 === 0) && (year % 100 !== 0)) || (year % 400 === 0);
            return isLeapYear ? 'Leap year' : 'Not a Leap year';
        }
        return year > 0 ? 'Plese enter valid year' : '';
    }

    return (
        <div>
            <div>
                Enter year : <input 
                    type='number'
                    placeholder='Enter a year' 
                    value={state.year}
                    onChange={handleInputChange} />
            </div>
            <label>{checkLeapYear(state.year)}</label>
        </div>
    );
}

export default LeapYear;