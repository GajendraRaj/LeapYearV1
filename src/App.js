import React from 'react';
import './App.css';
import LeapYear from './components/LeapYear'; 

function App() {
  return (
    <div className="App">
        <LeapYear />
    </div>
  );
}

export default App;
