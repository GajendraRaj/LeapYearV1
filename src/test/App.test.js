import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import LeapYear from '../components/LeapYear';

describe('App Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should load LeapYear component', () => {
    expect(wrapper.find(LeapYear)).toBeDefined();
  })

});
