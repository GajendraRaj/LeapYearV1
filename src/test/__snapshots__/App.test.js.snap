// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render correctly 1`] = `ShallowWrapper {}`;

exports[`App Component should render correctly 1`] = `ShallowWrapper {}`;
